### PR TITLE
Adds CheckedList, CheckedSelectbox and data controller

### PR DESCRIPTION
**Pull request by johnspackman**
CheckedList (and CheckedSelectBox) are easier to use than multiple selection, because checkboxes cannot be accidentally turned off so easily and have a more appropriate appearance.  This PR also includes a controller for binding

### CODE REVIEW

File: source/class/qx/data/controller/CheckedList.js [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+   Authors:
+     * John Spackman (github.com/johnspackman)
+
+************************************************************************ */
+
+/**
+ * Extension of `qx.data.controller.List` which adds support for `qx.ui.form.CheckedList`
+ * and `qx.ui.form.CheckedSelectBox`.
+ *
+ * The principal is thart the underlying `List` controller implementation has a model which
```
Comment (derrell) [inline source/class/qx/data/controller/CheckedList.js:23]: typo: `thart` vs `that`

File: source/class/qx/data/controller/CheckedList.js [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+ */
+qx.Class.define("qx.data.controller.CheckedList", {
+  extend: qx.data.controller.List,
+  
+  /**
+   * Constructor
+   *
+   * @param model {qx.data.Array?} the model array
+   * @param widget {qx.ui.core.Widget?} the widget target
+   * @param path {String} the path in the model for the caption
```
Comment (derrell) [inline source/class/qx/data/controller/CheckedList.js:38]: `path` is optional. Both `widget` and `path` default to null. I think these should be `qx.ui.core.Widget?null` and `String?null` respectively, as in the superclass:

```
   * @param target {qx.ui.core.Widget?null} The widget which should show the
   *   ListItems.
   *
   * @param labelPath {String?null} If the model contains objects, the labelPath
   *   is the path reference to the property in these objects which should be
   *   shown as label.
```

File: source/class/qx/data/controller/CheckedList.js [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+      this.__inUpdateChecked = true;
+      try {
+        // Maps of the widget item, indexed by the hashcode of the model item      
+        let children = {};
+        let toUncheck = {};
+        
+        target.getChildren().forEach(item => {
+          let itemModel = item.getModel();
+          let hash = itemModel ? itemModel.toHashCode() : "null";
+          children[hash] = item;
```
Comment (derrell) [inline source/class/qx/data/controller/CheckedList.js:317]: It appears possible for multiple items to hash to "null" thus overwriting items in the children map. Is this confirmed to be acceptable here?

**Review by derrell — CHANGES_REQUESTED** [review]
Looks useful!

**Review by cboulanger — APPROVED** [review]
Another great addition. I assume you'll add examples to the Demo Browser and Widget Browser (also important so that Theme authors can check that their theme works with the new widgets).

**Review by oetiker — APPROVED** [review]

**Review by hkollmann — APPROVED** [review]

**Review by level420 — APPROVED** [review]

**Review by derrell — APPROVED** [review]

**Pull request merged**